enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,445 @@
+```
+
+Other Pod annotation is `v1.multus-cni.io/default-network` which overrides default CNI from Multus'
+configuration file (created on DaemonSet start). This annotation is optional and its absence means
+Pod's default network will be ovn-kubernetes.
+
+Information about the Pod's interfaces are reported back by Multus also as an annotation `k8s.v1.cni.cncf.io/network-status`.
+Example below shows annotations of a Pod with two `bridge` CNI interfaces:
+```yaml
+Annotations:      k8s.v1.cni.cncf.io/network-status:
+                    [{
+                        "name": "ovn-kubernetes",
+                        "interface": "eth0",
+                        "ips": [
+                            "10.42.0.14"
+                        ],
+                        "mac": "0a:58:0a:2a:00:0e",
+                        "default": true,
+                        "dns": {}
+                    },{
+                        "name": "default/bridge-conf",
+                        "interface": "net1",
+                        "ips": [
+                            "192.168.20.202"
+                        ],
+                        "mac": "b2:6a:5c:ba:34:9a",
+                        "dns": {},
+                        "gateway": [
+                            "\u003cnil\u003e"
+                        ]
+                    },{
+                        "name": "default/bridge-conf",
+                        "interface": "net2",
+                        "ips": [
+                            "192.168.20.203"
+                        ],
+                        "mac": "7a:ca:cf:19:64:e8",
+                        "dns": {},
+                        "gateway": [
+                            "\u003cnil\u003e"
+                        ]
+                    }]
+                  k8s.v1.cni.cncf.io/networks: bridge-conf,bridge-conf
+```
+
+There is an annotation for NetworkAttachmentDefinition - `k8s.v1.cni.cncf.io/resourceName`.
+It is used when some CNI requires information about specific device that is prepared by device plugin.
+It is primarily used with SR-IOV and as such out of scope for this enhancement mentioned only for completeness.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+First, it must be noted that Multus CNI itself is a meta-CNI. From a high level perspective, its
+purpose is to call other CNIs according to the CNI configs supplied by the user in form of
+NetworkAttachmentDefinitions and Pod annotations.
+Any specific actions are related to the delegate CNIs. For example: creating veth pair,
+attaching one end of veth to the bridge and making the other end available within the Pod is
+responsibility of `bridge` CNI is this example.
+MicroShift team will create tests for CNIs that will be declared as supported. However these tests
+will not explore the breadth and depth of possible network setups, so ultimately the responsibility
+for correctness of the configuration is up to the user.
+
+#### Manifests
```

Comment:
Do we need to change ovnk manifest to copy/write the ovn-k8s-overlay-cni/cni-config-file to different directories? (crio may pick up the wrong CNI config file if ovnk keeps writing the file to default crio cni config directory, this may happen as ovnk uses hostnetwork, it doesn't have to wait for multus to start)

Reply:
`microshift-multus` will have a drop-in config for CRI-O that will explicitly set `cni_default_network` to multus, so CRI-O will be forced to wait for multus.
In other words, it will override [this setting](https://github.com/openshift/microshift/blob/main/packaging/crio.conf.d/11-microshift-ovn.conf#L4)

Reply:
When multus drop-in config is installed, will it take effect immediately. I remember having to restart crio in a test env, but not sure if it is necessary.

Reply:
Yes. But as mentioned in other comments and added to the enhancement, we'll instruct users to reboot the host after installing multus